addons/weave/2.5.2/install.sh

```diff
@@ -57,6 +57,17 @@ function weave_use_existing_network() {
     fi
 }
 
+function weave_health_check() {
+    if [[ -z $(kubectl get pods -n kube-system -l name=weave-net -o jsonpath="{.items[*].status.conditions[?(@.status!='True')]}") ]]; then
```

Comment:
What about `kubectl get pods -n kube-system -l name=weave-net -o jsonpath="{range .items[*]}{range .status.conditions[*]}{ .type }={ .status }{'\n'}{end}{end} | grep Ready=False` to be more precise about what condition you're checking?

Reply:
Hm... that's actually  a very good idea. Didn't know `jsonpath` was that powerful.